pandas/core/groupby/groupby.py

```diff
@@ -764,7 +766,19 @@ def _try_cast(self, result, obj, numeric_only=False):
             dtype = obj.dtype
 
         if not is_scalar(result):
-            if is_extension_array_dtype(dtype):
+            if is_datetime64tz_dtype(dtype):
+                # GH 23683
+                # Prior results _may_ have been generated in UTC.
+                # Ensure we localize to UTC first before converting
+                # to the target timezone
+                try:
+                    result = obj._values._from_sequence(result)
+                    result = result.tz_localize('UTC').tz_convert(dtype.tz)
+                except TypeError:
```

Comment:
pass a dtype to DTA._from_sequence is equivalent to localizing right? I think this is true for DTI 

cc @jbrockmendel @TomAugspurger

Reply:
It _should_, but I don't think DTA._from_sequence coerces `'UTC'` to a dtype downstream:

```
-> result = obj._values._from_sequence(result, dtype='UTC')
(Pdb) n
TypeError: data type 'UTC' not understood
```

Reply:
hmm this is a bug can u create an issue